qutip/core/coefficient.py

```diff
@@ -127,18 +129,6 @@ def shift(coeff, _t0=0):
     return ShiftCoefficient(coeff, _t0)
 
 
-def reduce(coeff, args):
-    """ Reduce decorated string coefficient to 1 object:
-    c = coefficient("t")
-    optimize(c + conj(c)) => coefficient("t+conj(t)")
-    """
-    reduced = coeff.optstr()
-    if reduced:
-        return coefficient(reduced, args=args)
-    else:
-        return coeff
-
-
```

Comment:
It's probably good this is going - because of the scoping rules of `Coefficient`, these optimisations would have produced incorrect results if the `args` lists weren't guaranteed to be identical.  As it stands, you could have something like
```python
base = qutip.coefficient('A * exp(w * t)', args={'A': 0, 'w': 0})
series = sum(base.replace(args={'A': A, 'w': w}) for A, w in [(1, 2*np.pi), (2, 4*np.pi), ...])
```
and then
```python
reduce(series, <anything>)
```
wouldn't work correctly.  That type of coefficient reuse (though not the summation) is something I do quite heavily in my own research code, because it lets me have a single compilation at the start of the programme, then I use the same compilation unit for several different variables in the same Hamiltonian without recompiling anythin.  Gives me a big speed boost, and a really nice feature of the new coefficient system!